dev_envs/mac-env-setup.md

```diff
@@ -0,0 +1,260 @@
+your code formatting will match the rest of the codebase
+  - Note: the pre-commit hooks will automatically run `black` on commit
+
+#### Configure `git` ####
+
+To allow access to the saved OSX Keychain credential for command line
+usage, run:
+
+```bash
+git config --global credential.helper osxkeychain
+```
+
+#### Set up commit signing with GPG ####
+
+Generate a key to sign your git commits and add it to your `~/.gitconfig`:
+
+```bash
+gpg --gen-key
+```
+
+Follow the prompts for name and email address, using either your CISA
+or Trio email address. The output should look like:
+
+```bash
+We need to generate a lot of random bytes. It is a good idea to perform
+some other action (type on the keyboard, move the mouse, utilize the
+disks) during the prime generation; this gives the random number
+generator a better chance to gain enough entropy.
+gpg: key EXAMPLE1234ABCDE marked as ultimately trusted
+gpg: revocation certificate stored as '/Users/username/path/to/.gnupg'
+public and secret key created and signed.
+
+pub   rsa3072 YYYY-MM-DD [SC] [expires: YYYY-MM-DD]
+      <fingerprint characters>
+uid                      User Name <user@example.com>
+sub   rsa3072 YYYY-MM-DD [E] [expires: YYYY-MM-DD]
+```
+
+The alphanumeric string from the key generation output line that says
+`gpg: key EXAMPLE1234ABCDE marked as ultimately trusted` is used as
+your **signing key**.
+
+Now that your new key has been generated, add it to your `~/.gitconfig`:
+
+1. Fill your `name` and `email`, using the same values as above
+1. Copy the alphanumeric string from the key generation output line that says
+    `gpg: key EXAMPLE1234ABCDE marked as ultimately trusted`
+1. Fill the alphanumeric signing key string in the `[user]` section:
+    `signingkey = EXAMPLE1234ABCDE`
+
+Next, export the public key via command line and add it to your GitHub account:
+
+1. `gpg --armor --export user@example.com`
+1. Copy the output, starting from `-----BEGIN PGP PUBLIC KEY BLOCK-----`
+1. Add a new PGP key to your [GitHub keys page](https://github.com/settings/keys)
+1. Paste the public key into the dialog and `Add GPG key`
+1. Confirm your password to continue (if requested)
+
+Lastly, enable the global configuration option for commit signing:
+
+```bash
+git config --global commit.gpgsign true
+```
+
+#### Generate `ssh` key to use git on the command line ####
+
+Reference: [Github: Generating a new SSH key](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent)
+
+Note: use `ed25519` instead of `rsa`
+
+```bash
+ssh-keygen -t ed25519
+```
+
+1. Save the generated key in the default location (e.g. `~/.ssh/id_ed25519`)
+1. Use a passphrase you'll remember and/or save in your keychain
+1. Start the `ssh-agent` in the background and add your new credential:
+
+```bash
+eval "$(ssh-agent -s)"
+ssh-add -K ~/.ssh/id_ed25519
+```
```

Comment:
Some indenting will make this code block part of the last bullet:
```suggestion
    ```bash
    eval "$(ssh-agent -s)"
    ssh-add -K ~/.ssh/id_ed25519
    ```
```

Reply:
I wish I'd seen the "Commit suggestion" button - I've fixed this in a848650149665216a068543dd1b50962d7ec2737